lib/Echidna/Exec.hs

```diff
@@ -63,11 +63,10 @@ vmExcept e = throwM $ case VMFailure e of {Illegal -> IllegalExec e; _ -> Unknow
 -- | Given an error handler, an execution function, and a transaction, execute that transaction
 -- using the given execution strategy, handling errors with the given handler.
 execTxWith :: (MonadState x m, Has VM x) => (Error -> m ()) -> m VMResult -> Tx -> m VMResult
-execTxWith h m t = do og <- get
-                      setupTx t
+execTxWith h m t = do setupTx t
                       res <- m
                       case (res, isRight $ t ^. call) of
-                        (Reversion,   _)         -> put og
+                        (Reversion,   _)         -> return ()
```

Comment:
I'm not sure this is correct. I think that if we make any state changes before encountering a revert, this will keep them. We can maybe do `put og >> result .= Just Reversion` to be more correct instead?

Reply:
You are right. I think I fixed it. Also, @agroce should be aware of this, since echidna could execute incorrectly some code before a revert. For instance this will fail:

```
contract C { 
  bool state = true;

  function f() public {
    state = false;
    revert();
  }

 function echidna_state() public returns (bool) {
    return state;
 }
}
```

Make sure you fix it and re-run your experiments.